android/capacitor/src/main/java/com/getcapacitor/Bridge.java

```diff
@@ -511,7 +509,7 @@ public PluginHandle getPluginWithRequestCode(int requestCode) {
                 }
             }
 
-            if (permissionRequestCode == requestCode) {
+            if (permissionRequestCode != null && permissionRequestCode == requestCode) {
                 return handle;
             }
```

Comment:
I would move this up into the legacy annotation block and then define `permissionRequestCode` as a local variable